src/Index.js

```diff
@@ -213,6 +213,35 @@ Index.prototype.saveObjects = function(objects, callback) {
   });
 };
 
+Index.prototype.replaceAllObjects = function(objects, opts, callback) {
+  if (typeof opts === 'function') {
+    callback = opts;
+    opts = {};
+  } else if (opts === undefined) {
+    opts = {};
+  }
+
+  var safe = opts.safe || false;
+  var tmpIndexName = this.indexName + '_' + Math.random() + '_tmp';
+
+  var copyResponse = this.as.copyIndex(this.indexName, tmpIndexName, ['settings', 'rules', 'synonyms']);
```

Comment:
This is not the way promises works unfortunately. Have a look at the old deleteBy implementation which is similar to what you want to accomplish: a meta method that calls other methods: https://github.com/algolia/algoliasearch-client-javascript/blob/develop/src/Index.js#L297

Reply:
I will dig into this, thank you :)